src/Guide/Markdown.hs

```diff
@@ -139,6 +140,10 @@ stringify = T.concat . map go
       HTML_BLOCK _      -> ""
       HTML_INLINE _     -> ""
 
+-- | Extract the first paragraph.
+extractSummary :: [MD.Node] -> MarkdownBlock
```

Comment:
This will extract the first heading if there isn't a paragraph preceding it. Instead you should parse Markdown with `toMarkdownBlockWithTOC` and extract all content before the first section ([`preface`](http://hackage.haskell.org/package/cmark-sections-0.1.0.3/docs/CMark-Sections.html#v:preface) from `mdTree`).